File: Algorithms/Java/selection_sort.java
public class selection_sort {
    public static void sel_sort(int arr[]){
        for(int i = 0;i<arr.length;i++){
            int min = arr[i],index = i;
            for(int j = i;j<arr.length;j++){
                if(min>arr[j]){
                    index = j;
                    min = arr[j];
                }
            }
            int temp = arr[i];
            arr[i] = min;
            arr[index] = temp;
        }
    }
    public static void main(String[] args){
        int arr[] = {7,45,82,5,72,6,36,8};
        // merge(arr, 0, arr, arr.length-1);
        sel_sort(arr);
        for(int i = 0;i<arr.length;i++){
            System.out.println(arr[i]);
        }
    }
}

File: Algorithms/C/selection_sort.c
#include <stdio.h>

void selection_sort(int arr[], int size)
{
    int i, j, temp, min;
    for (i = 0; i < size - 1; i++)
    {
        min = i;
        for (j = i; j < size; j++)
        {
            if (arr[j] < arr[min])
            {
                min = j;
            }
        }
        temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;
    }
}

int main()
{
    int arr[] = {5, 4, 3, 2, 1, 6, 7, 8, 9, 10, 0};
    int len = sizeof(arr) / sizeof(arr[0]);
    selection_sort(arr, len);
    int i;
    for (i = 0; i < len; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
    return 0;
}
